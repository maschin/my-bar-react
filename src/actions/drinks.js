import demoDrinks from '../tests/fixtures/drinks';

//ADD DRINK
export const addDrink = (drink) => ({
    type: 'ADD_DRINK',
    drink
});

export const startAddDrink = (drink = {}) => {
    return (dispatch, getState) => {
        drink.id = '123'+Date.now();
        dispatch(addDrink(drink));
    }
};

//EDIT DRINK
export const editDrink = (id, updates) => ({
    type: 'EDIT_DRINK',
    id,
    updates
});

export const startEditDrink = (id, updates) => {
    return (dispatch, getState) => {
        dispatch(editDrink(id, updates));
    }
};

//REMOVE DRINK
export const removeDrink = ( { id } = {}) => ({
    type: 'REMOVE_DRINK',
    id
});

export const startRemoveDrink = ( { id } = {}) => {
    return (dispatch, getState) => {
        dispatch(removeDrink({ id }));
    }
};

//SET DRINKS
export const setDrinks = (drinks) => ({
    type: 'SET_DRINKS',
    drinks
});

export const startSetDrinks = () => {
    return (dispatch, getState) => {
        dispatch(setDrinks(demoDrinks));
    }
}