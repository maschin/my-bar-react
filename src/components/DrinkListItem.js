import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Flag from 'react-world-flags';

export const DrinkListItem = ({
                                  id, title,
                                  category, vol,
                                  alc, price, aged, madeIn,
                                  description,
                                  isOpen, isEmpty,
                                  boughtAt, bottledAt, bestBefore,
                              }) => (
    <div>
        <Link className={`list-item ${isOpen ? 'list-item--opened' : ''} ${isEmpty ? 'list-item--empty' : ''}`} to={`/view/${id}`}>
            <div className={'list-item__title'}>
                {title}
            </div>
            <div className={'list-item__type'}>
                <Flag code={madeIn ? madeIn : ''} fallback={ <span>Unknown</span> } height="32" />&nbsp;
                <div>{category}</div>
            </div>
            <ul className={'list-item__params'}>
                <li className={'list-item__param__item'}>{vol}L</li>
                <li className={'list-item__param__item'}>{alc}%</li>
                <li className={'list-item__param__item'}>{numeral(price/100).format('$0,0.00')}</li>
                <li className={'list-item__param__item'}>
                    <FontAwesomeIcon icon={['fal', 'clock']} />&nbsp;
                    {aged ? aged : 'N/A'}
                </li>
            </ul>
            <ul className={'list-item__date'}>
                <li className={'list-item__date__item'}>
                    <FontAwesomeIcon icon={['fal', 'shopping-cart']} />&nbsp;
                    {moment(boughtAt).format('MMMM Do, YYYY')}
                </li>
                <li className={'list-item__date__item'}>
                    <FontAwesomeIcon icon="wine-bottle" />&nbsp;
                    {moment(bottledAt).format('MMMM Do, YYYY')}
                </li>
                <li className={'list-item__date__item'}>
                    <FontAwesomeIcon icon="flag-checkered" />&nbsp;
                    {moment(bestBefore).format('MMMM Do, YYYY')}
                </li>
            </ul>
            <div className={'list-item__description'}>
                {description}
            </div>
        </Link>
    </div>
);

export default DrinkListItem;