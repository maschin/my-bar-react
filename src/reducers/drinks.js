const drinksReducerDefaultState = [
    {
        id: 'sdfsdf',
        title: 'My bottle',
        category: 'Beer',
        vol: 0.5,
        alc: 5.6,
        price: 450,
        aged: 12,
        boughtAt: 123123,
        bottledAt: 12312312,
        bestBefore: 12312312,

    },
    {
        id: 'sdfsdf',
        title: 'My bottle',
        category: 'Beer',
        vol: 0.5,
        alc: 5.6,
        price: 450,
        aged: 12,
        boughtAt: 123123,
        bottledAt: 12312312,
        bestBefore: 12312312,

    }
];

export default (state = drinksReducerDefaultState, action) => {
    switch (action.type){
        case 'SET_DRINKS':
            return action.drinks;
        default:
    }       return state;
}