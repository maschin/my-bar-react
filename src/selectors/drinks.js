//Get visible drinks

export default (drinks, { text, sortBy }) => {
    return drinks.filter((drink) => {
        const textMatch = drink.title.trim().toLowerCase().includes(text.trim().toLowerCase()) || drink.category.trim().toLowerCase().includes(text.trim().toLowerCase());

        return textMatch;
    }).sort((a, b) => {
        switch(sortBy){
            default:
                return a.createdAt < b.createdAt ? 1 : -1;
        }
    })
}
