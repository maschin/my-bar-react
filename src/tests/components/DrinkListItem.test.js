import React from 'react';
import { shallow } from 'enzyme';
import DrinkListItem from '../../components/DrinkListItem';
import drinks from '../fixtures/drinks';

test('should render single item correctly', () => {
    const wrapper = shallow(<DrinkListItem key={drinks[1].id} {...drinks[1]}/>);
    expect(wrapper).toMatchSnapshot();
});