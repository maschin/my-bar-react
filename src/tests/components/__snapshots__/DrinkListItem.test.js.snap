// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render single item correctly 1`] = `
<div>
  <Link
    className="list-item  "
    replace={false}
    to="/view/laphroaigh"
  >
    <div
      className="list-item__title"
    >
      Laphroaig QA Cask Double Matured
    </div>
    <div
      className="list-item__type"
    >
      <Component
        code=""
        fallback={
          <span>
            Unknown
          </span>
        }
        height="32"
      />
       
      <div>
        Whisky
      </div>
    </div>
    <ul
      className="list-item__params"
    >
      <li
        className="list-item__param__item"
      >
        1
        L
      </li>
      <li
        className="list-item__param__item"
      >
        40
        %
      </li>
      <li
        className="list-item__param__item"
      >
        $48.04
      </li>
      <li
        className="list-item__param__item"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "clock",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        N/A
      </li>
    </ul>
    <ul
      className="list-item__date"
    >
      <li
        className="list-item__date__item"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fal",
              "shopping-cart",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        January 1st, 1968
      </li>
      <li
        className="list-item__date__item"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="wine-bottle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        January 1st, 1964
      </li>
      <li
        className="list-item__date__item"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="flag-checkered"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        November 1st, 1970
      </li>
    </ul>
    <div
      className="list-item__description"
    />
  </Link>
</div>
`;
